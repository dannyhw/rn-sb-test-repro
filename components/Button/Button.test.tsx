import '@testing-library/react-native/extend-expect';
import * as React from 'react';
import {render, screen} from '@testing-library/react-native';
import {MyButton} from './Button';

import {composeStory} from '@storybook/react';
import Meta, {Basic} from './Button.stories'; //ðŸ‘ˆ Our stories imported here.

test('examples of some things', async () => {
  render(
    <MyButton
      onPress={jest.fn()}
      text="testing"
      color="purple"
      textColor="white"
    />,
  );

  screen.getByText('testing');

  expect(screen.toJSON()).toMatchSnapshot();
});

const BasicButton = composeStory(Basic, Meta);

test('Checks if the form is valid', () => {
  render(<BasicButton />);

  screen.getByText('Hello World');
});
