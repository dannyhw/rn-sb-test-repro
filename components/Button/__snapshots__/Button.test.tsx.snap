// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples of some things 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
      "flex": 1,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignSelf": "flex-start",
        "backgroundColor": "purple",
        "borderRadius": 4,
        "flexGrow": 0,
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
          },
          {
            "color": "white",
          },
        ]
      }
    >
      testing
    </Text>
  </View>
</View>
`;
